src/common.h

```diff
@@ -243,8 +243,6 @@ struct [[gnu::unused]] noncopyable_t {
     noncopyable_t() = default;
     noncopyable_t(noncopyable_t &&) = default;
     noncopyable_t &operator=(noncopyable_t &&) = default;
-    noncopyable_t(const noncopyable_t &) = delete;
-    noncopyable_t &operator=(const noncopyable_t &) = delete;
```

Comment:
https://google.github.io/styleguide/cppguide.html#Copyable_Movable_Types says

> Explicitly declaring or deleting all four copy/move operations is permitted, but not required

but it also says

> a move-only class should explicitly declare the move operations, and a non-copyable/movable class should explicitly delete the copy operations

I read that as a preference for the explicit (albeit redundant) `delete`. I have no idea if there's a difference between "move-only" and "non-copyable/movable". If they're the same thing then the wording can be simplified.

Reply:
So in a case like this let's maybe disable the motivating clang-tidy check (which one is it?) to match the style guide.

Reply:
There's no clang-tidy check that complains about explicit constructors.

Reply:
Added them back.